opentelemetry-sdk/src/opentelemetry/sdk/metrics/__init__.py

```diff
@@ -218,7 +227,12 @@ def record_batch(
     ) -> None:
         """See `opentelemetry.metrics.Meter.record_batch`."""
         for metric, value in record_tuples:
-            metric.get_handle(label_values).update(value)
+            if isinstance(metric, Counter):
```

Comment:
Why did you remove the `update` methods? Now these ugly `isinstance` checks are required. Also, the `record` methods could probably be implemented in a base class when they could use `update`.

Reply:
I moved the `update` methods to the instruments themselves instead of the handles since `record_batch` works on the instruments. Also, when you are referring to the `record` methods being implemented in a base class, I think `add`, `set`, and `record` should be implemented separately in the specific metric instrument that uses them, not to be shared in a base `Metric` class.